EngineLayer/CrosslinkSearch/CrosslinkSearchEngine.cs

```diff
@@ -625,5 +674,19 @@ public static bool DissociationTypeGenerateSameTypeOfIons(DissociationType d, Di
             }
             return false;
         }
+
+        public static int BinarySearchGetIndex(double[] massArray, double targetMass)
```

Comment:
You can call the Binary search method from ModernSearchEngine. Remove this duplicate code.